chapters/en/slides/module3/module3_02.Rmd

```diff
@@ -0,0 +1,308 @@
+
+```{python}
+(alt.Chart(movies).transform_density(
+     'runtime',
+     groupby=['genre'],
+     as_=['runtime', 'density'],
+     steps=200)
+ .mark_area().encode(
+     x='runtime',
+     y='density:Q',
+     color='genre'))
+```
+
+Notes:
+To make the curves smoother,
+we can increase the resolution
+over which the KDE is calculated
+by setting the step size to a higher number.
+
+Now there will be 200 points calculated,
+which gives the area a smoother appearance.
+
+---
+
+## Making the overlapping areas slightly transparent make them easier to interpret
+
+```{python}
+(alt.Chart(movies).transform_density(
+     'runtime',
+     groupby=['genre'],
+     as_=['runtime', 'density'],
+     steps=200)
+ .mark_area(opacity=0.4).encode(
+     x='runtime',
+     y='density:Q',
+     color='genre'))
+```
+
+Notes:
+The density areas don't suffer the same issues as the histograms when made transparent.
+The continuous solid shape is easier to follow
+even when it is semi transparent
+overlaps with the other areas.
+
+The transparency gives us the advantage
+of knowing that there is not small density area hiding behind a bigger one.
+
+---
+
+## The y-axis does not indicate a meaningful value for density plots
+
+```{python}
+(alt.Chart(movies).transform_density(
+     'runtime',
+     groupby=['genre', 'country'],
+     as_=['runtime', 'density'])
+ .mark_area().encode(
+     x='runtime',
+     y='density:Q',
+     color='genre')
+ .properties(height=265).facet('country'))
+```
+
+Notes:
+If we facet these,
+you will notice that they look quite different from the faceted histograms
+in the previous slide deck.
+This is because the density rather than the count is shown by default on the y-axis.
+Showing the density means that it is easier for us to compare the shape of the distributions
+but is not really a useful numbers for us to know.
+
+---
+
+## Density plots of small datasets can be misleading
+
+```{python}
+(alt.Chart(movies[:10]).transform_density(
+    'runtime',
+    as_=['runtime', 'density'])  # Give a name to the KDE values, which we can use when plotting
+ .mark_area().encode(
+    x='runtime',
+    y='density:Q'))
+```
+
+Notes:
+Since density plots don't show the number of observations,
+they can be very misleading for small data sets
+as they will still appear smooth
+as if there were many data points to back up that smoothness.
+
+When looking at this plot,
+are you be able to tell that
+the density is made up of just ten observations?
+
+It is a little bit easier in Altair to tel this,
+because the default is to end the density where the data ends,
+which often creates sharp borders as in this plot.
+
+This is a great default in Altair,
+but many other plotting packages
+instead extend the density beyond the observations in the dataset
+to make it more smooth
+and it is therefore paramount
+to always ask how many observations there are
+before interpreting a density plot.
+
```

Comment:
Could adding a mark, like tick marks where each viz goes, be useful here to show that there are only 10 data points, and where they fall? 

This slide is also begging the question, would a histogram be better than a density in this small n situation?

Reply:
Good idea, let's add in another mark or add a subplot. One big concern is space on the slide.

I don't know the official answer to the second question, I would highlight that each come with their own caveats but those of the histogram might be more tolerable for a low n, I have a question on this in the lab